sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -120,6 +120,250 @@ public JsonWebKey(string keyId, string keyType, IList<string> keyOperations)
         /// HSM Token, used with "Bring Your Own Key"
         /// </summary>
         public byte[] T { get; set; }
+
+        internal override void ReadProperties(JsonElement json)
+        {
+            if (json.TryGetProperty("kid", out JsonElement kid))
+            {
+                KeyId = kid.GetString();
+            }
+            if (json.TryGetProperty("kty", out JsonElement kty))
+            {
+                KeyType = KeyTypeExtensions.ParseFromString(kty.GetString());
+            }
+            if (json.TryGetProperty("key_ops", out JsonElement key_ops))
+            {
+                foreach (var element in key_ops.EnumerateObject())
+                {
+                    KeyOps.Add(KeyOperationsExtensions.ParseFromString(element.Value.ToString()));
+                }
+            }
+            if (json.TryGetProperty("curveName", out JsonElement curveName))
+            {
+                CurveName = curveName.GetString();
+            }
+            if (json.TryGetProperty("n", out JsonElement n))
+            {
+                N = Encoding.ASCII.GetBytes(n.GetString());
```

Comment:
@bartonjs, @ahsonkhan, is there a way  to read Utf8/Ascii string bytes from JsonElement?

Reply:
Not directly.  For an amortized allocation-free you could make a new writer, call WriteAsValue on the element to write it into said writer, and then your bytes are output.Slice(1, length - 2) (get rid of the quotes).

Reply:
Looks like N is a base64 byte[]. The current approach will not give the right results if the user data was not encoded as a base64 string. The correct approach would be as follows:
```C#
N = n.GetBytesFromBase64();
``

Reply:
Thanks for the clarification. I will change it

Reply:
All these byte[] fields are actually base64url encoded. You can use the helper methods in Base64Url.cs to encode / decode them.

---
In reply to: [290088296](https://github.com/Azure/azure-sdk-for-net/pull/6457#discussion_r290088296) [](ancestors = 290088296)

Reply:
I see. Thanks Scott!